File: files/200-299/239/CH-239.py
import pandas as pd
import re

path = "200-299/239/CH-239 Extract from Text.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=5)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=10)

result = input.assign(
    **{'Product ID': input['Text'].apply(lambda x: re.findall(r'[a-zA-Z]{2}[0-9]{1}-[0-9]{2}', str(x)))}
).explode('Product ID').drop(
    columns='Text'
).reset_index(drop=True)

print(result.equals(test)) # True

File: files/200-299/239/CH-239.R
library(tidyverse)
library(readxl)

path = "files/200-299/239/CH-239 Extract from Text.xlsx"
input = read_excel(path, range = "B2:B7")
test = read_excel(path, range = "D2:D11")

result = input %>%
  mutate(
    `Product ID` = str_extract_all(Text, "[a-zA-Z]{2}[0-9]{1}-[0-9]{2}")
  ) %>%
  unnest(`Product ID`) %>%
  select(-Text)

all.equal(result, test)
# [1] TRUE
